### PR TITLE
Feature/Save Topic To CoreData

### PR DESCRIPTION
**Pull request by froggomad**
# [iOS - As a mobile user I can save data to my device](HTTPS://TRELLO.COM) 

## Changes
- Save Topic to CoreData
- Save Questions to CoreData
- Save ContextObjects to CoreData
- Added fetch methods to FetchController
- Added background context to CoreData

## Checklist
`choose one minimum`
- [x] created unit test
- [ ] updated unit test
- [ ] created UI test
- [ ] updated UI test
- [ ] bug fix/other

`required`
- [x] tested locally
- [x] updated the docs (https://github.com/Lambda-School-Labs/Labs26-Apollo-iOS-TeamA/blob/master/.github/documentation_standards.md)

`optional`
- [ ] added new dependencies

### CODE REVIEW

File: labs-ios-starter/Application/Network/ErrorHandler.swift [hunk 1]
```diff
@@ -54,7 +54,7 @@ class ErrorHandler {
             NetworkError.unauthorized.rawValue: UserError(title: "Unauthorized", message: "Please login again"),
             NetworkError.forbidden.rawValue: UserError(title: "Forbidden", message: "Access to that resource is restricted"),
             NetworkError.timeout.rawValue: UserError(title: "Timed out", message: "Please check your connection or try again later."),
-            NetworkError.notFound.rawValue: UserError(title: "Not Found", message: "Sorry, we're having trouble finding that resource")
+            NetworkError.notFound.rawValue: UserError(title: "Not Found", message: "Sorry, we're having trouble finding that resource"),
```
Comment (Shawn-James) [inline labs-ios-starter/Application/Network/ErrorHandler.swift:57]: Hmm.. did the linter add this comma at the end? We should probably fix that rule if so

File: labs-ios-starter/Surveys/Model Controllers/TopicController.swift [hunk 2]
```diff
@@ -5,29 +5,14 @@
 //  Created by Kenny on 9/12/20.
 //  Copyright © 2020 Lambda, Inc. All rights reserved.
 //
-
 import Foundation
-
+// MARK: - FOO -
```
Comment (Shawn-James) [inline labs-ios-starter/Surveys/Model Controllers/TopicController.swift:9]: The FOO has been marked

**Review by Shawn-James — APPROVED** [review]
Nice on the CoreData part. That linter was sure noisy on this one, but was great to get these files formatted.

File: labs-ios-starter/Application/Network/ErrorHandler.swift (hunk 1 above)
Comment (froggomad) [reply labs-ios-starter/Application/Network/ErrorHandler.swift:57]: Yeah, thought it was weird, and hated it for a minute, but then thought "oh, it's not causing a compiler error" so I just let it go. If we can get rid of it though, I think it looks cleaner without

File: labs-ios-starter/Surveys/Model Controllers/TopicController.swift (hunk 2 above)
Comment (froggomad) [reply labs-ios-starter/Surveys/Model Controllers/TopicController.swift:9]: Ha, was probably showing someone an example. Or maybe I just needed a metasyntactic variable used to represent an unspecified entity

File: labs-ios-starter/Application/Network/ErrorHandler.swift (hunk 1 above)
Comment (Shawn-James) [reply labs-ios-starter/Application/Network/ErrorHandler.swift:57]: Just changed it, it's the commas option. It's set to "always", should be "inline"

**Pull request merged**